script/bulk_import/generic_bulk.rb

```diff
@@ -859,14 +859,21 @@ def post_raw(row, group_names)
           name = user_full_name_from_id(user_id)
         end
 
+        if quote["post_id"]
+          topic_id = topic_id_from_imported_post_id(quote["post_id"])
+          post_number = post_number_from_imported_id(quote["post_id"])
+        end
+
+        bbcode_parts = []
+
+        bbcode_parts << name if name.present?
+        bbcode_parts << username if name.blank? && username.present?
+        bbcode_parts << "post:#{post_number}" if post_number.present?
+        bbcode_parts << "topic:#{topic_id}" if topic_id.present?
+        bbcode_parts << "username:#{username}" if username.present?
```

Comment:
Doesn't this create something like `[quote="foo, username:foo"]` if `name` is blank and `username` is "foo"?

How about replacing this whole block (line 867--876) with this?

```ruby
        bbcode =
          if username.blank? && name.blank?
            "[quote]"
          else
            bbcode_parts = []
            bbcode_parts << name.presence || username
            bbcode_parts << "post:#{post_number}" if post_number.present?
            bbcode_parts << "topic:#{topic_id}" if topic_id.present?
            bbcode_parts << "username:#{username}" if username.present? && name.present?

            %Q|[quote="#{bbcode_parts.join(", ")}"]|
          end
```

Reply:
> Doesn't this create something like [quote="foo, username:foo"] if name is blank and username is "foo"?

Yes, I assumed that was acceptable.

> How about replacing this whole block (line 867--876) with this?

Sure. Done!